DSL/transform.md

```diff
@@ -0,0 +1,364 @@
+# Transformation DSL
+
+This DSL (Domain-Specific Language) is inspired by JSON Patch and follows RFC 6901 for JSON Pointer representation in the path. It consists of an array of objects, each having the following keys:
+
+## Tranform fields
+
+| Property  | Description                                                      | Required | Data Type         |
+|-----------|------------------------------------------------------------------|----------|-------------------|
+| operation | Specifies the operation to perform                               | Yes      | String            |
+| path      | Specifies the path to the target in the schema                   | Yes      | Array of Strings  |
+| value     | Specifies the value to be used in the operation                  | No       | Any               |
+| from      | Specifies the source path for operations involving move/copy     | No       | Array of Strings  |
+| to        | Specifies the destination path for operations involving move/copy| No       | Array of Strings  |
+
+## Operations
+
+### 1. `append-property`
+
+Appends property in the provided `from` path to the `to` path. The `to` path should be an array, or it will be created as an empty array before appending.
+
+- **Takes**: `from`, `to`
+
+**Before**:
+
+```json
+{
+    "$schema": "https://json-schema.org/draft-03/schema",
+    "properties": {
+        "foo": {
+            "required": true
+        }
+    }
+}
+```
+
+**Transform**:
+
+```json
+[
+    { "operation": "append-property", "to": [ "required" ], "from": [ "properties", {} ] }
```

Comment:
Can you avoid `{}` in these examples, as they are meaningless without a corresponding condition?

Reply:
So, is it ok to just remove it and just say "from": [ "properties"]

Reply:
I think just have simpler examples that don't need `{}`, even if they are more fictitious

Reply:
Got it!